src/main/java/org/craftercms/studio/impl/v2/job/StudioClusterPublishedRepoSyncTask.java

```diff
@@ -401,29 +390,51 @@ private Set<String> getAllPublishingEnvironments(String site) {
     }
 
     private void updatePublishedBranch(String siteId, Git git, ClusterMember remoteNode, String branch)
-            throws CryptoException, GitAPIException, IOException, ServiceLayerException {
+            throws CryptoException, IOException, ServiceLayerException {
         logger.debug("Update published environment " + branch + " from " + remoteNode.getLocalAddress() +
                 " for site " + siteId);
         final Path tempKey = Files.createTempFile(UUID.randomUUID().toString(), ".tmp");
         Repository repo = git.getRepository();
         Ref ref = repo.exactRef(Constants.R_HEADS + branch);
         boolean createBranch = (ref == null);
 
-        logger.debug("Checkout " + branch);
-        CheckoutCommand checkoutCommand = git.checkout()
-                .setName(branch)
-                .setCreateBranch(createBranch);
-        if (createBranch) {
-            checkoutCommand.setStartPoint(remoteNode.getGitRemoteName() + "/" + branch);
-        }
-        checkoutCommand.call();
+        try {
+            logger.debug("Checkout " + branch);
+            CheckoutCommand checkoutCommand = git.checkout()
+                    .setName(branch)
+                    .setCreateBranch(createBranch);
+            if (createBranch) {
+                checkoutCommand.setStartPoint(remoteNode.getGitRemoteName() + "/" + branch);
+            }
+            checkoutCommand.call();
+            CheckoutResult result = checkoutCommand.getResult();
+            // If the checkout didn't throw an exception but still failed
+            if (result.getStatus() != CheckoutResult.Status.OK) {
+                // Throw an exception to trigger the cleanup
+                throw new IllegalStateException("Checkout failed for site " + siteId + " branch " + branch +
```

Comment:
Where would the clean-up happen?

Reply:
In the same catch block, I forgot to add this exception there too